### PR TITLE
Cleanup old / unused code

### PR DESCRIPTION
**Pull request by BenTheElder**
~depends on #1039~ rebased after merge

### CODE REVIEW

**Comment by BenTheElder** [comment]
eliminates the CRI package and related code, old serialization logic, etc.
follow up to the provider PR(s).

File: pkg/build/node/internal/container/docker/run.go [hunk 1]
```diff
@@ -17,77 +17,17 @@ limitations under the License.
...
-	}
-	// convert mounts to container run args
-	runArgs := o.RunArgs
-	for _, mount := range o.Mounts {
-		runArgs = append(runArgs, generateMountBindings(mount)...)
-	}
-	for _, portMapping := range o.PortMappings {
-		runArgs = append(runArgs, generatePortMappings(portMapping)...)
-	}
+func Run(image string, runArgs []string, containerArgs []string) error {
```
Comment (aojea) [inline pkg/build/node/internal/container/docker/run.go]: can we consolidate runArgs and containerArgs?

Comment (BenTheElder) [reply pkg/build/node/internal/container/docker/run.go]: no, they are different things.

container args are args to the container. runArgs are options to docker run itself. they must come before and after the image respectively.

Comment (BenTheElder) [reply pkg/build/node/internal/container/docker/run.go]: `docker run <flags> <image> <container args>`

**Pull request merged**